File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/enums/TradeItOrderAction.java
package it.trade.android.sdk.enums;

public enum TradeItOrderAction {
    BUY("buy"),
    SELL("sell"),
    BUY_TO_COVER("buyToCover"),
    SELL_SHORT("sellShort");

    private String actionValue;

    TradeItOrderAction(String actionValue) {
        this.actionValue = actionValue;
    }

    public String getActionValue() {
        return actionValue;
    }

    public static TradeItOrderAction getActionForValue(String actionValue) {
        for (TradeItOrderAction action: TradeItOrderAction.values()) {
            if (action.actionValue.equals(actionValue)) {
                return action;
            }
        }
        return null;
    }
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/enums/TradeItOrderAction.kt
package it.trade.android.sdk.enums

enum class TradeItOrderAction private constructor(val actionValue: String) {
    BUY("buy"),
    SELL("sell"),
    BUY_TO_COVER("buyToCover"),
    SELL_SHORT("sellShort");


    companion object {

        fun getActionForValue(actionValue: String): TradeItOrderAction? {
            for (action in TradeItOrderAction.values()) {
                if (action.actionValue == actionValue) {
                    return action
                }
            }
            return null
        }
    }
}
